### PR TITLE
Making Kafka consumer configurable for Kafka extract

### PR DESCRIPTION
**Pull request by shirshanka**
Global configuration for Kafka consumer for Kafka extracts. 

@hibikir This generalizes the proposal in https://github.com/linkedin/gobblin/pull/1059/

While making the changes, I also discovered that the socketTimeout was being set to the same value as the fetchTimeout in the old code. This resulted in intermittent failures in Kafka extraction when the amount of data in the source topic is less than the default fetchBytes. This was also commonly reported on the mailing lists by users trying out Kafka ingest locally. 

@abti, @ibuenros : Please review!

### CODE REVIEW

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaWrapper.java [hunk 1]
```diff
@@ -58,6 +61,10 @@
  */
 public class KafkaWrapper implements Closeable {
 
+
```
Comment (abti) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaWrapper.java]: Please delete the extra spacing

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaWrapper.java [hunk 2]
```diff
@@ -162,16 +182,56 @@ public void close() throws IOException {
...
+    public static final String CONFIG_KAFKA_FETCH_TOPIC_NUM_TRIES = CONFIG_PREFIX + "fetchTopicNumTries";
+    private static final int CONFIG_KAFKA_FETCH_TOPIC_NUM_TRIES_DEFAULT = 3;
+    public static final String CONFIG_KAFKA_FETCH_OFFSET_NUM_TRIES = CONFIG_PREFIX + "fetchOffsetNumTries";
+    private static final int CONFIG_KAFKA_FETCH_OFFSET_NUM_TRIES_DEFAULT = 3;
+
+    private final int socketTimeoutMillis;
+    private final int bufferSize;
+    private final String clientName;
+    private final int fetchCorrelationId;
+    private final int fetchTimeoutMillis;
```
Comment (abti) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaWrapper.java]: Millis generally go well with long

**Comment by abti** [comment]
Minor comments, rest LGTM

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaWrapper.java (hunk 2 above)
Comment (shirshanka) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaWrapper.java]: The Kafka API-s ask for int, so kept it that way.

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaWrapper.java (hunk 1 above)
Comment (shirshanka) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaWrapper.java]: Fixed!

**Comment by shirshanka** [comment]
Thanks for the quick review!

**Comment by aitchnu** [comment]
ConfigUtils.propertiesToConfig() method throws exception if two keys are prefixes of one another  which is true in case of fork in state.getProperties().

**Pull request merged**